core/blockchain.go

```diff
@@ -159,23 +160,82 @@ func (self *BlockChain) getProcInterrupt() bool {
 	return atomic.LoadInt32(&self.procInterrupt) == 1
 }
 
+func (self *BlockChain) blockIsGenesis(b *types.Block) bool {
+	return reflect.DeepEqual(b, self.Genesis())
+}
+
+// blockIsUnhealthy sanity checks for a block's health.
+// If header number is 0 and hash is not genesis hash.
+// Something is wrong; possibly missing/malformed header.
+func (self *BlockChain) blockIsUnhealthy(b *types.Block) error {
+	if b == nil {
+		return fmt.Errorf("nil block")
```

Comment:
We can move some of these to be initialized as variables once at start time ie
```go
var (
   errNilBlock    = errors.New("nil block")
   errNilHeader = errors.New("nil header")
)
```
and then use those variables, otherwise for every invocation that's extra function overhead :)